sql/2022/ecommerce/top_jslibs_by_device.sql

```diff
@@ -0,0 +1,47 @@
+#standardSQL
+# Top Javascript libraries used in Ecommerce sites
+SELECT
+  _TABLE_SUFFIX AS client,
+  app,
+  COUNT(DISTINCT url) AS freq,
+  total,
+  COUNT(DISTINCT url) / total AS pct
+FROM
+  `httparchive.technologies.2022_06_01_*`
+JOIN (
+  SELECT
+    _TABLE_SUFFIX,
+    url
+  FROM
+    `httparchive.technologies.2022_06_01_*`
+  WHERE
+    category = 'Ecommerce'
```

Comment:
I think you need a `DISTINCT` and to exclude Cart and GA EE to make the totals (and hence the percentages) to make sense. Currently you have 126% of sites using jQuery because of this.

```suggestion
  SELECT DISTINCT
    _TABLE_SUFFIX,
    url
  FROM
    `httparchive.technologies.2022_06_01_*`
  WHERE
    category = 'Ecommerce' AND
    (
      app != 'Cart Functionality' AND
      app != 'Google Analytics Enhanced eCommerce'
    )
```

Reply:
Thanks for noticing the incorrect calc. This just needed the app filters add the DISTINCT is unnecessary.